### ISSUE TITLE
I don't obtain the model save_epoch_12

### ISSUE THREAD

**Issue by KillerSH** [JoshuaPiinRueyPan/ViolenceDetection#17]
During training I get 2 warnings:
![er1](https://user-images.githubusercontent.com/17825161/54075572-a2380f80-42a9-11e9-8bf0-d294ba168a8c.PNG)

and

![er2](https://user-images.githubusercontent.com/17825161/54075580-af54fe80-42a9-11e9-9559-7966fda0f221.PNG)

I don't know if this is the reason why I don't obtain the directory save_epoch_12.
After training I don't obtain any save_epoch_12.
In TrainSettings.py I set as you said PATH_TO_SAVE_MODEL and the only thing I obtain in that location are 3 directories: test, train and validation.
Test is empty.
Train contains:
![1](https://user-images.githubusercontent.com/17825161/54075653-7406ff80-42aa-11e9-9430-00ebc8a8b978.png)

Validation contains:
![2](https://user-images.githubusercontent.com/17825161/54075663-8719cf80-42aa-11e9-892b-4806046b5014.png)

The following is my training log:
###############################################

WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.

 Using Network:  src.net.G2D19_P2OF_ResHB_1LSTM

2019-03-17 17:39:19.011373: W tensorflow/core/graph/graph_constructor.cc:1272] Importing a graph with a lower producer version 24 into an existing graph with producer version 27. Shape inference will have run different parts of the graph with different producer versions.
WARNING:tensorflow:From C:\Users\stefa\Anaconda3\envs\DeepLearning\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
In OpticalFlow:
         pool2.shape =  (?, 112, 112, 32)
         after transpose, out.shape =  (?, 112, 112, 32, 2)
         before Conv2, out.shape =  (?, 112, 112, 64)
         before Fc, out.shape =  (?, 28, 28, 64)
WARNING:tensorflow:From C:\Users\stefa\Downloads\ViolenceDetection-master\ViolenceDetection-master\src\net\G2D19_P2OF_ResHB_1LSTM.py:217: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
         Fc final.shape =  (?, 1024)
opticalFlowOut.shape =  (?, 1024)
In CNN:
         darknetOutput.shape =  (?, 7, 7, 1024)
         after transpose, out.shape =  (?, 7, 7, 1024, 2)
         before ConcatConv, out.shape =  (?, 7, 7, 2048)
         before Fc, out.shape =  (?, 7, 7, 128)
         Fc out.shape =  (?, 1024)
In Concat:
         before Fc, out.shape =  (?, 2048)
before LSTM, shape =  (?, ?, 1024)
WARNING:tensorflow:From C:\Users\stefa\Downloads\ViolenceDetection-master\ViolenceDetection-master\src\layers\RNN.py:14: LSTMCell.__init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is equivalent as tf.keras.layers.LSTMCell, and will be replaced by that in Tensorflow 2.0.
WARNING:tensorflow:From C:\Users\stefa\Downloads\ViolenceDetection-master\ViolenceDetection-master\src\layers\RNN.py:27: dynamic_rnn (from tensorflow.python.ops.rnn) is deprecated and will be removed in a future version.
Instructions for updating:
Please use `keras.layers.RNN(cell)`, which is equivalent to this API

WARNING:tensorflow:From C:\Users\stefa\Downloads\ViolenceDetection-master\ViolenceDetection-master\src\Classifier.py:34: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.

Reading Training set...
WARNING:tensorflow:From C:\Users\stefa\Anaconda3\envs\DeepLearning\lib\site-packages\tensorflow\python\ops\math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
         Done.

Reading Validation set...
         Done.

Reading Test set...
         Done.

2019-03-17 17:40:15.224395: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX
2019-03-17 17:40:16.114667: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties:
name: GeForce GT 710M major: 2 minor: 1 memoryClockRate(GHz): 1.55
pciBusID: 0000:01:00.0
totalMemory: 1.00GiB freeMemory: 823.06MiB
2019-03-17 17:40:16.130714: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1483] Ignoring visible gpu device (device: 0, name: GeForce GT 710M, pci bus id: 0000:01:00.0, compute capability: 2.1) with Cuda compute capability 2.1. The minimum required Cuda capability is 3.0.
2019-03-17 17:40:16.147362: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-17 17:40:16.155323: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0
2019-03-17 17:40:16.162249: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N
Path to save mode:  C:\Users\stefa\Downloads\ViolenceDetection-master\ViolenceDetection-master\Model

Start Training...

2019-03-17 17:41:13.257704: W tensorflow/core/framework/allocator.cc:124] Allocation of 2055208960 exceeds 10% of system memory.
2019-03-17 17:41:36.646958: W tensorflow/core/framework/allocator.cc:124] Allocation of 2055208960 exceeds 10% of system memory.
2019-03-17 17:41:51.127135: W tensorflow/core/framework/allocator.cc:124] Allocation of 1027604480 exceeds 10% of system memory.
2019-03-17 17:42:58.710324: W tensorflow/core/framework/allocator.cc:124] Allocation of 1027604480 exceeds 10% of system memory.
2019-03-17 17:45:50.314111: W tensorflow/core/framework/allocator.cc:124] Allocation of 2055208960 exceeds 10% of system memory.

################################################## end

I used the hockey dataset for training and I have:
CPU: I7 - 3537U dual core 2.2 GHz
GPU: NVidia Geforce 710M , VRAM: 1 GB
RAM: 8 GB

**Comment by JoshuaPiinRueyPan** [comment]
It seems that your Graphic Card does not have enough memory.
I use NV GTX 1080 Ti, and while training it consumes nearly 10 GB memory...
